src/Microsoft.AspNet.Mvc.ModelBinding/Binders/DictionaryModelBinder.cs

```diff
@@ -3,37 +3,15 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     public class DictionaryModelBinder<TKey, TValue> : CollectionModelBinder<KeyValuePair<TKey, TValue>>
     {
-        protected override bool CreateOrReplaceCollection(ModelBindingContext bindingContext,
-                                                          IList<KeyValuePair<TKey, TValue>> newCollection)
+        protected override object CreateModel(IEnumerable<KeyValuePair<TKey, TValue>> newCollection)
         {
-            CreateOrReplaceDictionary(bindingContext, newCollection, () => new Dictionary<TKey, TValue>());
-            return true;
-        }
-
-        private static void CreateOrReplaceDictionary(ModelBindingContext bindingContext,
-                                                      IEnumerable<KeyValuePair<TKey, TValue>> incomingElements,
-                                                      Func<IDictionary<TKey, TValue>> creator)
-        {
-            var dictionary = bindingContext.Model as IDictionary<TKey, TValue>;
-            if (dictionary == null || dictionary.IsReadOnly)
-            {
-                dictionary = creator();
-                bindingContext.Model = dictionary;
-            }
-
-            dictionary.Clear();
-            foreach (var element in incomingElements)
-            {
-                if (element.Key != null)
-                {
-                    dictionary[element.Key] = element.Value;
-                }
-            }
+            return newCollection.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
Cool. I'll take a second look at these changes. It seemed right to me, just wanted to know if it was required to change or just desirable.